decidim-core/app/controllers/concerns/decidim/action_authorization.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require "active_support/concern"
+
+module Decidim
+  module ActionAuthorization
+    extend ActiveSupport::Concern
+
+    included do
+      helper_method :authorize_action_path, :action_authorization
+    end
+
+    def authorize_action!(action_name, redirect_url: nil)
+      @action_authorizations ||= {}
+      @action_authorizations[action_name] = action_authorizer(action_name).authorize
+      status = @action_authorizations[action_name]
+
+      return if status.ok?
+      raise "Unauthorized" if status.code == :invalid
```

Comment:
We should raise a custom excfeption and rescue it at NeedsAuthorization like the CanCan one